tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2015-2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from datetime import timedelta
+from c7n.filters import Filter
+from c7n_azure.utils import Math
+from c7n.exceptions import PolicyValidationError
+
+
+class MetricFilter(Filter):
+
+    '''
+    Available intervals
+    PT1M, PT5M, PT15M, PT30M, PT1H, PT6H, PT12H, P1D
+
+    Available aggregations
+    Total, Average
+    '''
+
+    DEFAULT_TIMEFRAME = 24
+    DEFAULT_INTERVAL = 'P1D'
+    DEFAULT_AGGREGATION = 'Average'
+
+    aggregation_funcs = {
+        'Average': Math.mean,
+        'Total': Math.sum
+    }
+
+    def __init__(self, data, manager=None):
+        super(MetricFilter, self).__init__(data, manager)
+        # Metric name as defined by Azure SDK
+        self.metric = self.data.get('metric')
+        # gt (>), ge (>=), eq (==), le (<=), lt (<)
+        self.op = self.data.get('op')
+        # Value to compare metric value with self.op
+        self.threshold = self.data.get('threshold')
+        # Number of hours from current UTC time
+        self.timeframe = float(self.data.get('timeframe', self.DEFAULT_TIMEFRAME))
+        # Interval as defined by Azure SDK
+        self.interval = self.data.get('interval', self.DEFAULT_INTERVAL)
+        # Aggregation as defined by Azure SDK
+        self.aggregation = self.data.get('aggregation', self.DEFAULT_AGGREGATION)
+        # Aggregation function to be used locally
+        self.func = self.aggregation_funcs[self.aggregation]
+
+    def validate(self):
+        # Give appropriate error messages for missing information
+        if self.metric is None:
```

Comment:
could just define these as part of the jsonschema for this filter, else missing test coverage on a lot of this method.

Reply:
@tbarlow12 you can reference the schema on `class ValueFilter(Filter):` in c7n\filters\core.py as it has many similarities.  Since your validation is simple you might be able to push it all off on that.